Week9/MAKEME.md

```diff
@@ -1,101 +1,36 @@
-
-> Let's continue to learn a little more about scope and Closures. 
-
-
-2. What will be the output of the following code - and more importantly - WHY? 
-
-```js
-for (var i = 0; i < 3; i++) {
-      setTimeout(function() { alert(i); }, 1000 + i);
-}
 ```
-
-
-3. Write a function that would allow you to do this:
-```js
-let addSix = createBase(6);
-addSix(10); // returns 16
-addSix(21); // returns 27
-```
-
-4. You will need to create an HTML document out of the below snippet to run the below code. A hint - the code is syntactically correct but doesn't do what you would expect. Can you see why and fix it?
-
-Don't cheat - but if you get stuck ... http://stackoverflow.com/questions/750486/javascript-closure-inside-loops-simple-practical-example
-
-```html 
-<button id="btn-0">Button 1!</button>
-<button id="btn-1">Button 2!</button>
-<button id="btn-2">Button 3!</button>
-
-<script type="text/javascript">
-    
-    let prizes = ['A Unicorn!', 'A Hug!', 'Fresh Laundry!'];
-    for (let btnNum = 0; btnNum < prizes.length; btnNum++) {
-        // for each of our buttons, when the user clicks it...
-        document.getElementById('btn-' + btnNum).onclick = function() {
-            // tell her what she's won!
-            alert(prizes[btnNum]);
-        };
-    }
-</script>
+Topics discussed this week:
+• (re)writing data structures (in JSON)
+• Promises
 ```
 
-5. Rewrite the code below to Async: 
-
-```js
-1.
-
-let sum = calculateSum(2, 6);
-console.log(sum);
-
-2.
+## Step 1: Read
 
-let results = $.getJSON('http://myapi.com');
-showResults(results);
+- If you need to refresh your memory on es6 classes: [es6-classes-in-depth](https://ponyfoo.com/articles/es6-classes-in-depth)
+- Also read this article on scopes & closures: [explaining-javascript-scope-and-closures](https://robertnyman.com/2008/10/09/explaining-javascript-scope-and-closures/)
 
-3.
-
-let sum = calculateSum(2, 6);
-if (sum > 8) {
-    console.log('larger than 8');
-}
-
-4.
+## Step 2: Feedback
 
-let data = $.getJSON('http://myapi.com');
-data = data.map(function (x) { return x * 8; });
+- Create at least 2 issues (bug / feature / code improvement) on another teams github repository. Do this in pairs. 
+-  Solve the issue proposed by another students in your github repo. More info [here](https://hackyourfuture.slack.com/files/michahell/F31BX1XT6/Merging_a_local_branch_into_master)
 
-writeDataToFile(data);
-```
+## Step 3: Pair programming promises challenge 
 
+This week you will work in a team!
 
-## Step 2: Feedback
+So please:
 
-- Create at least 2 issues (bug / feature / code improvement) on another teams github repository. Do this in pairs. 
--  Solve the issue proposed by another students in your github repo. More info [here](https://hackyourfuture.slack.com/files/michahell/F31BX1XT6/Merging_a_local_branch_into_master)
+- (one of your team mates should) Fork this repository: https://github.com/HackYourFuture/PromisessObjectsAndStringRendering
```

Comment:
This repo uses a newable object using prototype definitions. Do we want to use ES6 classes instead?